tests/console/validate_packages_and_patterns.pm

```diff
@@ -27,10 +27,39 @@ if (check_var('VALIDATE_PCM_PATTERN', 'azure')) {
         installed => 1,
         condition => sub { check_var_array('PATTERNS', 'azure') },
         pattern   => 1,
-        packages  => [qw(azuremetadata cloud-regionsrv-client docker-img-store-setup growpart
-              patterns-public-cloud-Microsoft-Azure python-azure-agent python-azurectl
-              python-susepubliccloudinfo regionServiceClientConfigAzure
-              supportutils-plugin-suse-public-cloud)],
+        packages  => [qw(azuremetadata
+              cloud-regionsrv-client docker-img-store-setup growpart supportutils-plugin-suse-public-cloud
+              patterns-public-cloud-Microsoft-Azure python-azure-agent python-azurectl python-susepubliccloudinfo
+              regionServiceClientConfigAzure
+              )],
+    };
+}
+elsif (check_var('VALIDATE_PCM_PATTERN', 'aws')) {
+    # We have different pattern names on SLE 15 and SLE 12
+    my $aws_pattern = is_sle('15+') ? 'Amazon_Web_Services' : 'Amazon-Web-Services';
+    $software{$aws_pattern} = {
+        repo      => 'Module-Public-Cloud',
+        installed => 1,
+        condition => sub { check_var_array('PATTERNS', 'aws') },
+        pattern   => 1,
+        packages  => is_sle('15+') ? [qw(
```

Comment:
Two things, here, please file a bug reporting the difference. Secondly we could define common part and then push additional packages to the array depending on the version not to have same packages twice.